lib/make/makefile.blast

```diff
@@ -31,11 +42,50 @@ MAKECALL_BLAST_NCBI_DBS    = echo "	Called with path to NCBI dbs: $(NCBI_BLAST_D
 MAKECALL_BLAST_INFILES     = echo "	Input files: $^ (`ls -lL $^|tr '\n' ','`)" >> $@.makecall
 MAKECALL_BLAST             = $(MAKECALL_BLAST_VERSION); $(MAKECALL_BLAST_NCBI_DBS); $(MAKECALL_BLAST_INFILES)
 
+# BLASTn-specific macros
+#
+# *Don't redefine!*
+MAKECALL_BLASTn_VERSION     = echo "$$(date +"%Y%m%d %H:%M:%S"): $@ was made with $$(blastn -version|grep 'blastn:')" > $@.makecall
+MAKECALL_BLASTn_PARAMS    = echo "	Called with parameters: $(BLASTn_OPTS)" >> $@.makecall
+
+MAKECALL_BLASTn_DB  = echo "  BLAST database: $(BLAST_NUCLEOTIDE_DB_PATH)/$(BLAST_NUCLEOTIDE_DB), timestamp: $(ls -lL $(BLAST_NUCLEOTIDE_DB_PATH)/$(BLAST_NUCLEOTIDE_DB).blastdb)" >> $@.makecall
+MAKECALL_BLAST_INFILES     = echo "	Input files: $^ (`ls -lL $^|tr '\n' ','`)" >> $@.makecall
+MAKECALL_BLASTn             = $(MAKECALL_BLASTn_VERSION); $(MAKECALL_BLASTn_PARAMS); $(MAKECALL_BLASTn_DB); $(MAKECALL_BLAST_INFILES)
+
+# BLASTx-specific macros
+#
+# *Don't redefine!*
+MAKECALL_BLASTx_VERSION     = echo "$$(date +"%Y%m%d %H:%M:%S"): $@ was made with $$(blastx -version|grep 'blastx:')" > $@.makecall
+MAKECALL_BLASTx_PARAMS    = echo "	Called with parameters: $(BLASTx_OPTS)" >> $@.makecall
+
+MAKECALL_BLASTx_DB  = echo "  BLAST database: $(BLAST_PROTEIN_DB_PATH)/$(BLAST_PROTEIN_DB), timestamp: $(ls -lL $(BLAST_PROTEIN_DB_PATH)/$(BLAST_PROTEIN_DB).blastdb)" >> $@.makecall
+MAKECALL_BLAST_INFILES     = echo "	Input files: $^ (`ls -lL $^|tr '\n' ','`)" >> $@.makecall
+MAKECALL_BLASTx             = $(MAKECALL_BLASTx_VERSION); $(MAKECALL_BLASTx_PARAMS); $(MAKECALL_BLASTx_DB); $(MAKECALL_BLAST_INFILES)
+
+
 # *** Targets ***
 
 # UNTESTED and assumes you've also included makefile.misc that can convert a fastq.gz to an fna:
 all_mega.blastn6s: $(subst .fna,.mega.blastn6.tsv.gz,$(wildcard *.fastq.gz))
```

Comment:
This will not work. You're trying to replace .fna in a list of files that ends with .fastq.gz.

Reply:
Good point, better stick with `$(wildcard *.fna)`